cmd/snap/cmd_export_key.go

```diff
@@ -73,7 +111,20 @@ func (x *cmdExportKey) Execute(args []string) error {
 			"name":                keyName,
 			"public-key-sha3-384": pubKey.ID(),
 			"since":               time.Now().UTC().Format(time.RFC3339),
-			// XXX: To support revocation, we need to check for matching known assertions and set a suitable revision if we find one.
+		}
+		if x.Revoke {
+			headers["until"] = headers["since"]
+			var authContext auth.AuthContext
+			sto := storeNew(nil, authContext)
+			// XXX A missing assertion isn't an error here,
+			// since we will just export a new one.  However, it
+			// isn't currently possible to distinguish between
+			// "assertion not found" and other errors at this
+			// point.
+			previousAssertion, _ := fetchAccountKeyAssertion(sto, pubKey)
+			if previousAssertion != nil {
```

Comment:
Besides the comment from @pedronis above, this also seems strange in that a revoke only makes sense if the key exists in the first place, which means it ought to have a revision.

Also, I believe we should we make a distinction between a revoke meaning "stop using it" vs. one saying "make all signatures invalid". I'm concerned people may end up doing the latter while they attempt to do the former, which would destroy all of their previously signed data unintendedly.